### PR TITLE
Allow configuration to be read from eclipse linked resources

### PR DESCRIPTION
**Pull request by alpapad**
If the tslint configuration is a en eclipse linked resource, the plugin can not read it, as it expects a filesystem path. 

Such a feature is needed in cases where the application, e2e, unit tests are hosted under the same source but with different typings and/or dependencies (which are managed by the build tool chain). In these cases, one can create nested eclipse projects with linked resources. An example project could have:

	<linkedResources>
		<link>
			<name>build</name>
			<type>2</type>
			<locationURI>PARENT-2-PROJECT_LOC/.build/app</locationURI>
		</link>
		<link>
			<name>tslint.json</name>
			<type>1</type>
			<locationURI>PARENT-2-PROJECT_LOC/tslint.json</locationURI>
		</link>
		<link>
			<name>typings</name>
			<type>2</type>
			<locationURI>virtual:/virtual</locationURI>
		</link>
		<link>
			<name>typings/app</name>
			<type>2</type>
			<locationURI>PARENT-2-PROJECT_LOC/src/typings/app</locationURI>
		</link>
	</linkedResources>

Here, PARENT-2-PROJECT_LOC/tslint.json is located in the parent project root folder.

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/eclipse-tslint](https://github.com/palantir/eclipse-tslint), @alpapad! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: com.palantir.tslint/.classpath [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
 	<classpathentry excluding="bridge/" kind="src" path="src"/>
 	<classpathentry kind="lib" path="lib/guava-14.0.1.jar" sourcepath="lib/guava-14.0.1-sources.jar"/>
-	<classpathentry kind="lib" path="lib/jackson-annotations-2.2.0.jar"/>
+	<classpathentry kind="lib" path="lib/jackson-annotations-2.2.0.jar" sourcepath="C:/Users/papadaka/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.2.0/jackson-annotations-2.2.0-sources.jar"/>
```
Comment (adidahiya) [inline com.palantir.tslint/.classpath]: this line change looks unrelated

File: com.palantir.tslint/src/com/palantir/tslint/Builder.java [hunk 2]
```diff
@@ -88,7 +88,7 @@ private void lint(IResource resource) throws IOException {
                 configurationPath = projectFile.getAbsolutePath();
             }
         } else {
-            configurationPath = project.getFile("tslint.json").getRawLocation().toOSString();
+            configurationPath = project.getFile("tslint.json").getLocation().toOSString();
```
Comment (adidahiya) [inline com.palantir.tslint/src/com/palantir/tslint/Builder.java]: does this mean L85 should change to use `getLocation()` as well?

File: com.palantir.tslint/.classpath (hunk 1 above)
Comment (alpapad) [inline com.palantir.tslint/.classpath]: You are right. I will update.

File: com.palantir.tslint/src/com/palantir/tslint/Builder.java (hunk 2 above)
Comment (alpapad) [inline com.palantir.tslint/src/com/palantir/tslint/Builder.java]: Yes, just in case someone wants to use the preferences with an eclipse relative path...

Comment (alpapad) [inline com.palantir.tslint/src/com/palantir/tslint/Builder.java]: Updated
